### DISCUSSION TITLE
How to restrict updates of a single image to their matched regexes?

### DISCUSSION THREAD

**Discussion by alecgerona** [renovatebot/renovate#18995] [Request Help] [answered]
Hi, I would like to know how to use Renovate to manage updates of a gitops k8s manifest repository that has multiple environments with different tag formats. 

For example I have a tag like master-4ca13e(commit short sha). This should be updated only if there's a newer tag that matches the `master-` regex. Same goes with normal semver tags such as 1.0.0; it should only be updated if there's a newer tag matching `X.y.z`

**Comment by rarkins** [comment, marked as answer]
Start by looking at the `allowedVersions` option in `packageRules`.

Alternatively it might need `extractVersions` or even `regex` versioning
